src/main/java/com/shopping_app/shoppingApp/service/AddressService.java

```diff
@@ -0,0 +1,93 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.NotFoundException;
+import com.shopping_app.shoppingApp.domain.Address;
+import com.shopping_app.shoppingApp.domain.User;
+import com.shopping_app.shoppingApp.model.Address.Request.AddressRequest;
+import com.shopping_app.shoppingApp.model.Address.Response.AddressResponse;
+import com.shopping_app.shoppingApp.repository.AddressRepository;
+import com.shopping_app.shoppingApp.utils.UserPrincipal;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class AddressService {
+
+    private final AddressRepository addressRepository;
+    private final UserService userService;
+
+    public AddressResponse addAddress(AddressRequest addressRequest) {
+        UserPrincipal principal = (UserPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Address address = convertToAddress(addressRequest);
+        User user = userService.fetchUserById(principal.getId());
+        address.setUser(user);
+        Address saveAddress = addressRepository.save(address);
+        return convertToAddressResponse(saveAddress);
+    }
+
+    public AddressResponse updateAddress(AddressRequest addressRequest, Long id) {
+        Address address = fetchAddressById(id);
+        address.setStreet(addressRequest.getStreet());
+        address.setState(addressRequest.getState());
+        address.setCountry(addressRequest.getCountry());
+        address.setCity(addressRequest.getCity());
+        address.setPincode(addressRequest.getPincode());
+        Address updatedAddress = addressRepository.save(address);
+        return convertToAddressResponse(updatedAddress);
+    }
+
+    public AddressResponse deleteAddress(Long id) {
```

Comment:
Why do we need to return an address here?

Reply:
actually Not needed , but by sending response, we can just confirm the deleted address , when testing from api level

Reply:
It's not good to violate some standard practices for the ease of testing. You can change it to return void

Reply:
Sure , replaced with no-content (void) type